src/components/Reservations.js

```diff
@@ -0,0 +1,77 @@
+import React, { useEffect, useContext } from 'react';
+import { useDispatch, useSelector } from 'react-redux';
+import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
+import { faBars } from '@fortawesome/fontawesome-free-solid';
+import { getReservations } from '../Redux/APIFunctions/reservations';
+import localStorageActions from '../utils/localStorage';
+import Navbar from './NavBar';
+import './CreateReservation.css';
+import Reservation from './Reservation';
+
+import MenuContext from '../Context/MenuContext';
+
+const Reservations = () => {
+  const {
+    reservations, loading, error,
+  } = useSelector((state) => state.reservations);
+  const { showSideBar, sideBar } = useContext(MenuContext);
+
+  const userObject = localStorageActions.getUser();
+
+  const dispatch = useDispatch();
+
+  useEffect(() => {
+    dispatch(getReservations());
+  }, [dispatch]);
+
+  if (reservations === undefined && loading) {
+    <h1 className="loading">Loading reservation ...</h1>;
+  }
+
+  if (error) {
+    <h1 className="error">Kindly refresh the page ...</h1>;
+  }
+
+  console.log(reservations);
+
+  return (
+
+    (
+      <div className="all-reservations">
+        {
+            sideBar ? <Navbar />
+              : (
+                <>
+                  <FontAwesomeIcon className="toggle" id="toggle" onClick={() => showSideBar()} icon={faBars} />
+
+                  <div className="reserve-cards">
+
+                    {reservations?.map((reservation) => (
+                      reservation.user_id === userObject.id
+                        ? (
+                          <div key={reservation.id}>
+                            <Reservation
+                              id={reservation.id}
+                              reason={reservation.reason}
+                              duration={reservation.duration}
+                              startDay={reservation.start_day}
+                              endDay={reservation.end_day}
+                              userID={reservation.user_id}
+                              hotelID={reservation.hotel_id}
+                            />
```

Comment:
- [ ] I strongly suggest you slice or truncate your dates, so they are more human readable when presenting:

![image](https://user-images.githubusercontent.com/57421823/193181109-ea706eb0-974a-493e-bf48-147e38bbfc03.png)

## Tips

### I managed to do it with the `slice()` method, this is how:

```JAVASCRIPT
                     <Reservation
                        id={reservation.id}
                        reason={reservation.reason}
                        duration={reservation.duration}
                        startDay={reservation.start_day.slice(0, 10)}
                        endDay={reservation.end_day.slice(0, 10)}
                        userID={reservation.user_id}
                        hotelID={reservation.hotel_id}
                      />
```
Result:

![image](https://user-images.githubusercontent.com/57421823/193181669-6345be7a-5d03-478a-98d1-88a2c0494c38.png)

### Also, you can add a space before each item, that way your reservations will look more beautiful:

```JAVASCRIPT
                          <div key={reservation.id}>
                            <Reservation
                              id={reservation.id}
                              reason={` ${reservation.reason}`}
                              duration={` ${reservation.duration}`}
                              startDay={` ${reservation.start_day.slice(0, 10)}`}
                              endDay={` ${reservation.end_day.slice(0, 10)}`}
                              userID={reservation.user_id}
                              hotelID={reservation.hotel_id}
                            />

                          </div>
```

Final result:

![image](https://user-images.githubusercontent.com/57421823/193182782-00c63c3a-5e99-444d-b0ef-dec465258e47.png)



Now that you know how to do it, go ahead and truncate those dates!  They are just strings. 💪